### PR TITLE
async-work: fix complete callback segv/ticks

### PR DESCRIPTION
**Pull request by legendecas**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included

### CODE REVIEW

File: src/napi/node_api_async.c [hunk 1]
```diff
@@ -57,6 +61,8 @@ static void iotjs_uv_work_after_cb(uv_work_t* req, int status) {
       jerry_release_value(jval_err);
     }
   }
+
+  iotjs_process_next_tick();
```
Comment (yorkie) [inline src/napi/node_api_async.c]: Please add a regression on https://github.com/Rokid/ShadowNode/blob/1a7a7649be26330cebde42131808142eaede5b3b/test/napi/napi_async.test.js :)

**Comment by yorkie** [comment]
Ah, we did skip the test_async/test.js at https://github.com/Rokid/ShadowNode/blob/1a7a7649be26330cebde42131808142eaede5b3b/test/napi-testsets.json#L15

@legendecas Could you please remove the skip flag now?

File: src/napi/node_api_async.c (hunk 1 above)
Comment (legendecas) [reply src/napi/node_api_async.c]: tests are already included https://github.com/Rokid/ShadowNode/blob/1a7a7649be26330cebde42131808142eaede5b3b/test/napi/napi_async.test.js#L12.
Yet it doesn't test as expected though.

**Comment by legendecas** [comment]
@yorkie it's not possible to enable `test_async/test.js` pulled from node repo for not implemented sync api
 https://github.com/Rokid/ShadowNode/blob/master/test/addons-napi/test_async/test.js#L17

**Comment by yorkie** [comment]
Or could we have the patches to N-API Test Suites just like the openwrt/buildroot package, which provides a way to cover more NTS without any forks with source?

@mhdawson @gabrielschulhof, I think this tooling could also be included into our future NTS repository, and it helps implementor to get more coverages.

**Comment by yorkie** [comment]
@legendecas we should avoid any case that uses freed block, so please set the pointer to be null after it gets freed at: https://github.com/Rokid/ShadowNode/blob/c56f4f96bd53259befece9f2ea996fe784f98a7e/src/napi/node_api_async.c#L101

And use `IOTJS_ALLOC` and `IOTJS_RELEASE`, too.

**Pull request merged**